app/Http/Controllers/V2/Terrafund/TerrafundCreateGeometryController.php

```diff
@@ -0,0 +1,757 @@
+    //       $perimeter += $distance;
+    //     }
+    //     $averageDistance = $perimeter / ($numVertices - 1);
+    //     $threshold = $averageDistance * 1.25; // Adjust this threshold as needed
+    //     for ($i = 0; $i < $numVertices - 1; $i++) {
+    //       $distance = $this->calculateDistance($coordinates[$i], $coordinates[$i + 1]);
+    //       if (abs($distance - $averageDistance) > $threshold) {
+    //         $spikes[] = $coordinates[$i];
+    //       }
+    //     }
+    //   }
+    //   return $spikes;
+    // }
+    public function detectSpikes($geometry)
+    {
+        $spikes = [];
+
+        if ($geometry['type'] === 'Polygon' || $geometry['type'] === 'MultiPolygon') {
+            $coordinates = $geometry['type'] === 'Polygon' ? $geometry['coordinates'][0] : $geometry['coordinates'][0][0]; // First ring of the polygon or the first polygon in the MultiPolygon
+            $numVertices = count($coordinates);
+            $totalDistance = 0;
+
+            for ($i = 0; $i < $numVertices - 1; $i++) {
+                $totalDistance += $this->calculateDistance($coordinates[$i], $coordinates[$i + 1]);
+            }
+
+            for ($i = 0; $i < $numVertices - 1; $i++) {
+                $distance1 = $this->calculateDistance($coordinates[$i], $coordinates[($i + 1) % $numVertices]);
+                $distance2 = $this->calculateDistance($coordinates[($i + 1) % $numVertices], $coordinates[($i + 2) % $numVertices]);
+                $combinedDistance = $distance1 + $distance2;
+
+                if ($combinedDistance > 0.6 * $totalDistance) {
+                    // Vertex and its adjacent vertices contribute more than 25% of the total boundary path distance
+                    $spikes[] = $coordinates[($i + 1) % $numVertices];
+                }
+            }
+        }
+
+        return $spikes;
+    }
+
+    public function insertCriteriaSite($POLYGON_ID, $CRITERIA_ID, $valid)
+    {
+        $criteriaSite = new CriteriaSite();
+        $criteriaSite->polygon_id = $POLYGON_ID;
+        $criteriaSite->criteria_id = $CRITERIA_ID;
+        $criteriaSite->uuid = Str::uuid();
+        $criteriaSite->valid = $valid;
+
+        try {
+            $criteriaSite->save();
+
+            return true;
+        } catch (\Exception $e) {
+            return $e->getMessage();
+        }
+    }
+
+    public function checkBoundarySegments(Request $request)
+    {
+        $uuid = $request->query('uuid');
+        $geometry = PolygonGeometry::where('uuid', $uuid)->first();
+
+        if (! $geometry) {
+            return response()->json(['error' => 'Geometry not found'], 404);
+        }
+        $geojson = DB::selectOne('SELECT ST_AsGeoJSON(geom) AS geojson FROM polygon_geometry WHERE uuid = :uuid', ['uuid' => $uuid])->geojson;
+        $geojsonArray = json_decode($geojson, true);
+        $spikes = $this->detectSpikes($geojsonArray);
+        $SPIKE_CRITERIA_ID = 8;
```

Comment:
What's the purpose of this magic number?

Reply:
I uploaded a table on the description of the PR, that is the number referenced to.